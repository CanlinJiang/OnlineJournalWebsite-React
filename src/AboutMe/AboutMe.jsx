import React from 'react'
import Frame from "../UtilitiesComponents/MainFrame/Frame"
import styles from "./AboutMe.module.css"



export default function AboutMe() {
    


    const BodyComponent = () => (
    <div className={styles.bodyContainer}>
        <div className={styles.textContainer}>
            <p>Hi, here is Canlin (Autumn) Jiang. </p>
            <p>Thank you for being here. This website is a demo of a private personal article collection, which is similar to blogs, but mainly for collection.</p>
            <p>Me? I'm a graduate student learning Information System, but an aweful photographer as well, as you can feel on this website</p>
            <p>Interest? Coding, writing, and picturing. They are basically the same thing to me, creating</p>
            <p>More? Add my instagram <a href="https://www.instagram.com/treee_autumnnn/">here</a></p>
        </div>
        <div className={styles.imageContainer}></div>
    </div>
    )

    return (
        <Frame BodyComponent={BodyComponent}/>

    )
}
