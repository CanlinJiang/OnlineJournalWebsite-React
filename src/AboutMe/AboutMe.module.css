.bodyContainer {
    display:grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    justify-content: center;
  
}

.textContainer {
    
    width: 100%;
    float:left;
    font-size: 18px;
    line-height: 1.6;
    text-align: justify;
}
.textContainer p{
    margin: 30px 0px;
}

.textContainer a{
    text-decoration: underline;
}

.textContainer a:hover{
    color: rgb(136, 130, 130);
}

.textContainer a:active{
    color: rgb(92, 133, 86);
}

.textContainer a:visited{
    color: black;
}

.imageContainer {
    
  width: 100%;
  height: 400px;
  background-image: url("/images/IMG_4205-900_x_900.jpeg");
  background-repeat: no-repeat;
  background-size: contain;
  background-position:center;
  
}

@media (max-width: 767px) {
    .bodyContainer{
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr 2fr;
        grid-gap: 10px;
        margin: 30px;
    }
    .imageContainer{
        order: -1;
        width: 100%;
        height: 100%;
        
        background-size: contain;
    }
    .textContainer{
        align-self: flex-start;
    }
}

@media (min-width: 768px) and (max-width: 1023px) {
    .bodyContainer{
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-gap: 20px;
    }
    
}

@media (min-width: 1024px) {
    .bodyContainer{
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-gap: 30px;
        width: 80%;
        justify-content: space-between;
        align-items: center;
        margin-left: auto;
        margin-right: auto;
        margin-bottom: 200px;
    }
    
    
}