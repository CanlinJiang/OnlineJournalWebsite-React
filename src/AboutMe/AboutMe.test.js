import AboutMe from "./AboutMe"
import React from "react";

describe("AboutMe tests", () => {
    it("renders correctly", () => {
        
      const { container } = render(<AboutMe />);
      expect(container).toMatchSnapshot();
    });
  
  });