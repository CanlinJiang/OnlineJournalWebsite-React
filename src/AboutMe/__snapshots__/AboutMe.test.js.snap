// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <nav
        class="navigation"
      >
        <ul
          class="leftBar"
        >
          <li>
            <a
              href="/minelist"
            >
              Mine's
            </a>
          </li>
          <li>
            <a
              href="/moodchoosing"
            >
              Moods
            </a>
          </li>
          <li>
            <a
              href="/randomwalk"
            >
              Random Walk
            </a>
          </li>
        </ul>
        <ul
          class="rightBar"
        >
          <li
            id="home"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            id="addNewArticle"
          >
            <a
              href="/addnewarticle"
            >
              Add New Article
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="body"
    >
      <div
        class="bodyContainer"
      >
        <div
          class="textContainer"
        >
          <p>
            Hi, here is Canlin (Autumn) Jiang. 
          </p>
          <p>
            Thank you for being here. This website is a demo of a private personal article collection, which is similar to blogs, but mainly for collection.
          </p>
          <p>
            Me? I'm a graduate student learning Information System, but an aweful photographer as well, as you can feel on this website
          </p>
          <p>
            Interest? Coding, writing, and picturing. They are basically the same thing to me, creating
          </p>
          <p>
            More? Add my instagram 
            <a
              href="https://www.instagram.com/treee_autumnnn/"
            >
              here
            </a>
          </p>
        </div>
        <div
          class="imageContainer"
        />
      </div>
    </div>
    <div
      class="footer"
    >
      <nav>
        <ul
          class="navigation"
        >
          <li
            id="aboutMe"
          >
            <a
              href="/aboutme"
            >
              About me
            </a>
          </li>
          <li
            id="contactMe"
          >
            <a
              href="/contactme"
            >
              Contact me
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
